modules/abstract-utxo/src/recovery/backupKeyRecovery.ts

```diff
@@ -54,23 +57,31 @@ export interface FormattedOfflineVaultTxInfo<TNumber extends number | bigint = n
 
 /**
  * This transforms the txInfo from recover into the format that offline-signing-tool expects
- * @param coinName
+ * @param coin
  * @param txInfo
  * @param txHex
  * @returns {{txHex: *, txInfo: {unspents: *}, feeInfo: {}, coin: void}}
  */
 function formatForOfflineVault<TNumber extends number | bigint = number>(
-  coinName: string,
+  coin: AbstractUtxoCoin,
   txInfo: OfflineVaultTxInfo<TNumber>,
   txHex: string
 ): FormattedOfflineVaultTxInfo<TNumber> {
+  let unspents;
+  if (coin.amountType === 'bigint') {
+    unspents = txInfo.inputs.map((input) => {
+      return { ...input, value: Number(input.value), valueString: input.value.toString() };
+    });
+  } else {
+    unspents = txInfo.inputs;
+  }
```

Comment:
Remove the branching, always put `valueString` on the unspents. This will simplify the return type as well.